client/client-multi/client-c/src/platform-impl/kaa_tcp_channel.c

```diff
@@ -0,0 +1,1349 @@
+kaa_error_t kaa_tcp_channel_sync_handler(void *context, const kaa_service_t services[], size_t service_count) {
+    KAA_RETURN_IF_NIL(context, KAA_ERR_BADPARAM);
+    kaa_tcp_channel_t *channel = (kaa_tcp_channel_t *) context;
+
+    kaa_error_t error_code = KAA_ERR_NONE;
+
+    if (services && service_count > 0) {
+        error_code = kaa_tcp_channel_update_pending_services(channel, services, service_count);
+        KAA_RETURN_IF_ERR(error_code);
+    }
+
+    return error_code;
+}
+
+/*
+ * Release Kaa TCP channel context
+ */
+kaa_error_t kaa_tcp_channel_release_context(void *context) {
+    KAA_RETURN_IF_NIL(context, KAA_ERR_BADPARAM);
+    kaa_tcp_channel_t *channel = (kaa_tcp_channel_t *) context;
+    kaa_error_t error_code = KAA_ERR_NONE;
+
+    if (channel->parser) {
+        KAA_FREE(channel->parser);
+        channel->parser = NULL;
+    }
+
+
+    kaa_tcp_channel_release_access_point(channel);
+
+    kaa_buffer_destroy(channel->in_buffer);
+
+    kaa_buffer_destroy(channel->out_buffer);
+
+
+    if (channel->pending_request_services) {
+        KAA_FREE(channel->pending_request_services);
+        channel->pending_request_services = NULL;
+    }
+
+    if (channel->encryption.aes_session_key) {
+        KAA_FREE(channel->encryption.aes_session_key);
+        channel->encryption.aes_session_key = NULL;
+        channel->encryption.aes_session_key_size = 0;
+    }
+
+    if (channel->encryption.signature) {
+        KAA_FREE(channel->encryption.signature);
+        channel->encryption.signature = NULL;
+        channel->encryption.signature_size = 0;
+    }
+
+    if (channel->supported_services) {
+        KAA_FREE(channel->supported_services);
+        channel->supported_services = NULL;
+        channel->supported_service_count = 0;
+    }
+
+    channel->access_point.state = AP_UNDEFINED;
+
+    KAA_FREE(context);
+
+
+    return error_code;
+}
+
+/*
+ * Init Kaa TCP channel transport context
+ */
+kaa_error_t kaa_tcp_channel_init(void *context, kaa_transport_context_t *transport_context)
+{
+    KAA_RETURN_IF_NIL2(context, transport_context, KAA_ERR_BADPARAM);
+    KAA_RETURN_IF_NIL2(transport_context->platform_protocol, transport_context->bootstrap_manager, KAA_ERR_BADPARAM);
+    kaa_tcp_channel_t *channel = (kaa_tcp_channel_t *) context;
+
+    channel->transport_context = *transport_context;
+
+    return KAA_ERR_NONE;
+}
+
+/*
+ * Set access point to Kaa TCP channel.
+ */
+kaa_error_t kaa_tcp_channel_set_access_point(void *context, kaa_access_point_t *access_point)
+{
+    KAA_RETURN_IF_NIL2(context, access_point, KAA_ERR_BADPARAM);
+    kaa_tcp_channel_t *channel = (kaa_tcp_channel_t *) context;
+
+    kaa_error_t error_code = KAA_ERR_NONE;
+
+    KAA_LOG_INFO(channel->logger, KAA_ERR_NONE, "Kaa TCP channel setting access point.");
+
+    if (channel->access_point.state != AP_NOT_SET) {
+        KAA_LOG_TRACE(channel->logger, KAA_ERR_NONE, "Kaa TCP channel removing previous access point (%d)", channel->access_point.id);
```

Comment:
1. I propose to use the [0x%08X] format to print out the access point id.
2. Please. replace %d with %u to print out the connection data length.